test-specifications/token-service/tokenDisscoiateTransaction.md

```diff
@@ -0,0 +1,142 @@
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "method": "dissociateToken",
+  "params": {
+    "accountId": "0.0.2533",
+    "tokenIds": [
+      "0.0.579680",
+      "0.0.90649"
+    ],
+    "commonTransactionParams": {
+      "signers": [
+        "302e020100300506032b65700422042031f8eb3e77a04ebe599c51570976053009e619414f26bdd39676a5d3b2782a1d"
+      ]
+    }
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "result": {
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Token IDs:**
+
+- The IDs of the tokens to dissociate.
+
+| Test no | Name                                                   | Input                                                                                                                                              | Expected response                                                                                    | Implemented (Y/N) |
+|---------|--------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Dissociates no tokens with an account                  | accountId=<CREATED_ACCOUNT_ID>, commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                                                    | The token dissociation succeeds and no disassociations are made.                                     | N                 |
+| 2       | Dissociates a token that doesn't exist with an account | accountId=<CREATED_ACCOUNT_ID>, tokenIds=["123.456.789"], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                          | The token dissociation fails with an INVALID_TOKEN_ID response code from the network.                | N                 |
+| 3       | Dissociates a token that is deleted with an account    | accountId=<CREATED_ACCOUNT_ID>, tokenIds=[<DELETED_TOKEN_ID>], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                     | The token dissociation fails with an TOKEN_WAS_DELETED response code from the network.               | N                 |
+| 4       | Dissociates a token that is empty with an account      | accountId=<CREATED_ACCOUNT_ID>, tokenIds=[""], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                                     | The token dissociation fails with an SDK internal error.                                             | N                 |
+| 5       | Dissociates a token twice with an account              | accountId=<CREATED_ACCOUNT_ID>, tokenIds=[<CREATED_TOKEN_ID>, <CREATED_TOKEN_ID>], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>] | The token dissociation fails with an TOKEN_ID_REPEATED_IN_TOKEN_LIST response code from the network. | N                 |
+
```

Comment:
It may be worthwhile to add the following cases:
1. Dissociate from 3 different token types, all have no balance and are associated to the account.
1. Dissociate from 3 different token types, one has a non-zero balance, the remainder are as above.
1. Dissociate from 3 different token types, one is not associated with the account, the remainder are as (1) above.
1. Dissociate from 3 different token types, one is `frozen` for this account, the remainder are as (1) above.
1. Dissociate from 3 different token types, one is `paused`, the remainder are as (1) above.

It may be worth looking at the various permutations of KYC flag as well, but I'm not as clear on that set of options.

Reply:
Added more test cases